storage/src/digest.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{fmt, io, ops::Deref};
+
+use smallvec::SmallVec;
+use snarkvm_utilities::{ToBytes, Write};
+
+type InnerType = SmallVec<[u8; 64]>;
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct Digest(pub InnerType);
```

Comment:
As far as I can tell, this is a type applicable to _any_ digest, while we're using the digest name pretty specifically in the codebase right now (for the tx ledger digest); I'd change this name to e.g. `AnyDigest` or `SomeDigest` (not great, but at least differentiates it from the ledger digest).

Reply:
we dont call vecs `AnyVec` or packets `AnyPacket` (in a generic context). the ledger digests also use `Digest`. this type is truely just a newtype over the smallvec -- we only use it for digests though (they are usually small). i've though about renaming it to something like `SmallX` -- don't know a good `X` though.

Reply:
What about alternatively changing the instances of `digest` to `ledger_digest`? I'm very ok with there being a "generic" `Digest` object, as long as we don't mistake it for some specific subset of it.

Reply:
yeah, that's generally the plan